### PR TITLE
DP-614: Added notification message if CH not found or if company already registered

### PR DESCRIPTION
**Pull request by dbgoaco**
[DP-614](https://noticingsystem.atlassian.net/jira/software/projects/DP/boards/22?selectedIssue=DP-614)

### CODE REVIEW

File: Frontend/CO.CDP.OrganisationApp/Pages/Shared/_FlashMessage.cshtml [hunk 1]
```diff
@@ -13,4 +13,18 @@
             <p class="govuk-notification-banner__heading">@(tempDataService.Get<string>(FlashMessageTypes.Success))</p>
         </div>
     </div>
+}
+
+@if (tempDataService.Peek<string>(FlashMessageTypes.Important) != null)
+{
+    <div class="govuk-notification-banner" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
```
Comment (andymantell) [inline Frontend/CO.CDP.OrganisationApp/Pages/Shared/_FlashMessage.cshtml:20]: Spacing is a bit tight above this message. Could you add the same top margin from the notification banner?

{7ECEBC74-77CE-44CE-AAA1-9EB70FD7C6FB}

File: Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouseApi.cs [hunk 2]
```diff
@@ -21,7 +24,7 @@ public class CompaniesHouseApi(IConfiguration configuration, ILogger<CompaniesHo
                 .GetAsync()
```
Comment (andymantell) [inline Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouseApi.cs:24]: If I search for a valid company number with a trailing whitespace it does not find it. Which is easily done - if you double click on the number on the companies house website in order to select and copy and paste it, it actually comes with a trailing space.

Could we trim the value before searching?

**Review by andymantell — CHANGES_REQUESTED** [review]

File: Frontend/CO.CDP.OrganisationApp/Pages/Shared/_FlashMessage.cshtml (hunk 1 above)
Comment (andymantell) [reply Frontend/CO.CDP.OrganisationApp/Pages/Shared/_FlashMessage.cshtml:20]:
```suggestion
    <div class="govuk-notification-banner govuk-!-margin-top-8" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
```

Comment (dbgoaco) [reply Frontend/CO.CDP.OrganisationApp/Pages/Shared/_FlashMessage.cshtml]: Good ideas, done both, thanks for the review.

File: Frontend/CO.CDP.OrganisationApp.Tests/ThirdPartyApiClients/CompaniesHouseApiTests.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        _mockConfiguration.Setup(c => c["CompaniesHouse:Password"]).Returns("testpassword");
+
+        using var httpTest = new HttpTest();
+        httpTest.RespondWithJson(expectedProfile);
+
+        // Act
+        var result = await _companiesHouseApi.GetProfile(companyNumber);
+
+        // Assert
+        Assert.NotNull(result);
```
Comment (dharmverma) [inline Frontend/CO.CDP.OrganisationApp.Tests/ThirdPartyApiClients/CompaniesHouseApiTests.cs:60]: As discussed earlier in dev meetings, please use FluentAssertion

File: Frontend/CO.CDP.OrganisationApp.Tests/ThirdPartyApiClients/CompaniesHouseApiTests.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        Assert.Equal(expectedAddress.AddressLine1, result.AddressLine1);
+    }
+
+    [Fact]
+    public async Task GetProfile_ShouldReturnCompanyProfile()
+    {
+        // Arrange
```
Comment (dharmverma) [inline Frontend/CO.CDP.OrganisationApp.Tests/ThirdPartyApiClients/CompaniesHouseApiTests.cs:46]: Removing these comments

File: Frontend/CO.CDP.OrganisationApp.Tests/ThirdPartyApiClients/CompaniesHouseApiTests.cs (hunk 3 above)
Comment (dbgoaco) [reply Frontend/CO.CDP.OrganisationApp.Tests/ThirdPartyApiClients/CompaniesHouseApiTests.cs:54]: Thanks for the review, updated.

File: Frontend/CO.CDP.OrganisationApp.Tests/ThirdPartyApiClients/CompaniesHouseApiTests.cs (hunk 4 above)
Comment (dbgoaco) [reply Frontend/CO.CDP.OrganisationApp.Tests/ThirdPartyApiClients/CompaniesHouseApiTests.cs]: Done.

**Review by dharmverma — APPROVED** [review]

File: Frontend/CO.CDP.OrganisationApp.Tests/ThirdPartyApiClients/CompaniesHouseApiTests.cs (hunk 3 above)
Comment (dbgoaco) [reply Frontend/CO.CDP.OrganisationApp.Tests/ThirdPartyApiClients/CompaniesHouseApiTests.cs]: Done.

**Pull request merged**